log/log.go

```diff
@@ -329,6 +337,12 @@ func SetFileName(name string) {
 	std.name = name
 }
 
+func SetOutput(out Output) {
+	std.rw.Lock()
+	defer std.rw.Unlock()
+	std.out = out
```

Comment:
The places that use "std.out" (access it) to print don't seem to be grabbing the reader-lock of std.rw. 
Actually nobody is doing std.rw.Rlock().  Probably tiny corner case as most users either don't change
these values or only do them at initialization in the single-threaded part of their code.  Just wondering 
about this lock's usage.

Reply:
right, I should use std.mu.Lock(), will make another pr to fix it